### PR TITLE
Add webhook trigger allowed_methods/local_only options

### PR DESCRIPTION
**Pull request by esev**
## Proposed change

Add Overflow menu for configuring allow_internet/allow_methods options on webhook triggers. These options are being added in https://github.com/home-assistant/core/pull/66494.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
automation:
  trigger:
    - platform: webhook
      webhook_id: "some_hook_id"
      allowed_methods:
        - POST
        - PUT
      local_only: true
```

Screenshot 2022-02-16 4 01 12 PM
Screenshot 2022-02-16 4 01 35 PM

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/65348
- This PR is related to issue or discussion: https://github.com/home-assistant/core/pull/66494
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/21637

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by balloob** [comment]
We cannot have 2 overflow menus. That looks weird.

**Comment by esev** [comment]
> We cannot have 2 overflow menus. That looks weird.

I think I misinterpreted your suggestion in https://github.com/home-assistant/core/pull/65928

> Although I would even move that into the overflow menu so that we don't have to confuse the user always with it.

Did you mean to include it into the overflow menu for the `ha-automation-trigger-row`? I agree that the double `mdiDotsVertical` icon for the overflow menu looks odd. What would you think about just switching the icon to `mdiCog`?

**Comment by balloob** [comment]
cog is definitely an improvement. Maybe @matthiasdebaat also has an idea for improvement.

**Comment by frenck** [comment]
Just a small consistency thingy maybe, we recently added a similar feature for  users:

CleanShot 2022-02-15 at 08 26 15@2x

In this case, it uses: Allow only local,  while this PR adds: Allow for internet.

Should we create consistency here and use same logic/wording? (As in toggle to allow local only, instead of allowing internet access).

File: src/panels/config/automation/trigger/types/ha-automation-trigger-webhook.ts [hunk 1]
```diff
@@ -70,44 +80,95 @@ export class HaWebhookTrigger extends LitElement {
...
+          <ha-icon-button
+            @click=${this._copyUrl}
+            slot="trailingIcon"
+            .label=${this.hass.localize(
+              "ui.panel.config.automation.editor.triggers.type.webhook.copy_url"
+            )}
+            .path=${mdiContentCopy}
+          ></ha-icon-button>
+        </ha-textfield>
+        <ha-icon-overflow-menu
```
Comment (bramkragten) [inline src/panels/config/automation/trigger/types/ha-automation-trigger-webhook.ts:138]: Use `ha-button-menu` with `ha-check-list-item`

Comment (bramkragten) [reply src/panels/config/automation/trigger/types/ha-automation-trigger-webhook.ts:138]: https://github.com/home-assistant/frontend/blob/dev/src/panels/config/devices/ha-config-devices-dashboard.ts#L433-L450

**Comment by frenck** [comment]
Backend has been merged.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**